src/react/HomePage.js

```diff
@@ -29,185 +29,136 @@ import {
-
-    const {
-      connectedDevices,
-      totalDevices,
-      appengineStatus,
-      realmManagementStatus,
-      pairingStatus,
-      flowStatus,
-      interfaces,
-      triggers
-    } = this.state;
-
-    return (
-      <Container fluid className="p-3">
-        <Row>
-          <Col xs={12}>
-            <h2 className="mb-4">Astarte Dashboard</h2>
+  const cellSpacingClass = "mb-3";
+
+  return (
+    <Container fluid className="p-3">
+      <Row>
+        <Col xs={12}>
+          <h2 className="mb-4">Astarte Dashboard</h2>
+        </Col>
+        <Col xs={6} className={cellSpacingClass}>
+          <ApiStatusCard
+            appengine={appengineStatus}
+            realmManagement={realmManagementStatus}
+            pairing={pairingStatus}
+            flow={flowStatus}
+          />
+        </Col>
+        <WaitForData status={deviceStatusStats}>
+          <Col xs={6} className={cellSpacingClass}>
+            <DevicesCard
+              connectedDevices={devicesStats.connected_devices}
+              totalDevices={devicesStats.total_devices}
+            />
           </Col>
+        </WaitForData>
+        <WaitForData status={interfacesStats}>
           <Col xs={6} className={cellSpacingClass}>
-            <ApiStatusCard
-              appengine={appengineStatus}
-              realmManagement={realmManagementStatus}
-              pairing={pairingStatus}
-              flow={flowStatus}
+            <InterfacesCard
+              interfaceList={interfaces}
+              onInterfaceClick={redirectToLastInterface}
+              onInstallInterfaceClick={() => { history.push("/interfaces/new") }}
             />
           </Col>
-          { isReady(connectedDevices) &&
-            <Col xs={6} className={cellSpacingClass}>
-              <DevicesCard
-                connectedDevices={connectedDevices}
-                totalDevices={totalDevices}
-              />
-            </Col>
-          }
-          { isReady(interfaces) &&
-            <Col xs={6} className={cellSpacingClass}>
-              <InterfacesCard
-                interfaceList={interfaces}
-                onInterfaceClick={this.redirectToLastInterface}
-                onInstallInterfaceClick={() => { this.props.history.push("/interfaces/new") }}
-              />
-            </Col>
-          }
-          { isReady(triggers) &&
-            <Col xs={6} className={cellSpacingClass}>
-              <TriggersCard
-                triggerList={triggers}
-                onInstallTriggerClick={() => { this.props.history.push("/triggers/new") }}
-              />
-            </Col>
-          }
-        </Row>
-      </Container>
-    );
-  }
+        </WaitForData>
+        <WaitForData status={triggersStats}>
+          <Col xs={6} className={cellSpacingClass}>
+            <TriggersCard
+              triggerList={triggers}
+              onInstallTriggerClick={() => { this.props.history.push("/triggers/new") }}
```

Comment:
`this.props.history` should be just `history`. Basically, since you refactored the component in a functional one, there should be no more references to a `this` variable in this file

Reply:
yea, i forgot to change it during the refactor